app/registries/foreman/plugin/fact_parser_registry.rb

```diff
@@ -0,0 +1,24 @@
+module Foreman
+  class Plugin
+    class FactParserRegistry
+      attr_reader :parsers
+
+      def initialize
+        @parsers = {}.with_indifferent_access
+      end
+
+      def parser_for(type)
+        @parsers[type]
+      end
+
+      def register(type, klass, default = false)
```

Comment:
Could you explain the default concept? Perhaps in a code comment. I don't really understand what it's trying to achieve and I realize you are only moving existing code. However, this is a good time to avoid mistakes from the past.

Reply:
I've added a docs. I hope it's describe all options.